Lib/_pydecimal.py

```diff
@@ -2131,10 +2164,11 @@ def _power_exact(self, other, p):
             else:
                 return None
 
-            if xc >= 10**p:
+            strxc = _convert_to_str(xc, p)
+            if strxc is None:
```

Comment:
Is this equivalent in context? `xc >= 10**p` and `_convert_to_str(xc, p) is None` aren't equivalent for example for `xc = 1000000` and `p = 3`, as `_convert_to_str` only checks for *any* power of 10, not for just the p-th power of 10. Are larger powers of 10 ruled out some other way?

Reply:
I believe larger powers are already ruled out, but am not 100% sure of that - the code is very involved.

I'm relying in the end on the stated _purpose_ of the "check exact" context, which is to determine whether the input is exactly representable in decimal floating point with `p` digits of precision. `10**i` for any `i >= 0` is, provided only `p >= 1`. I'm confident enough in that to skip adding a check that the string has exactly `p+1` digits too.

Reply:
I added a new comment to spell this out. The new behavior may or may not be right, and the old behavior may or may not be wrong. Had to take my best guess, because I've had little luck so far trying to out-think this code ☹️